content/post/formatting-with-style.md

```diff
@@ -0,0 +1,247 @@
+---
+author: m09
+date: 2018-10-12
+title: "Formatting. With Style."
+image: /post/formatting-with-style/teapot.jpg
+description: "Formatting code using existing styles."
+categories: ["technical", "MLonCode"]
+---
+
+Ensuring that a codebase is consistent in style is both hard and
+costly, yet it is extremely important for maintainability and to
+reduce technical debt. Why is it hard and costly?  There are many
+contributing factors but the main ones are certainly:
+
+- teams might not have set up a linting process early enough or at all
+- people handling the codebase might be changing on a regular basis
+
+To allow new or current team members to code in a consistent way, we
+therefore need to model existing style and use it to lint new code. At
+source{d}, we call that breakfast. Let's go!
+
+## Lookout
+
+This problem is one of the many painpoints we are currently tackling
+with [Lookout](https://github.com/src-d/lookout), our brand new and
+frankly awesome assisted code review framework.
+
+The purpose of Lookout is to bring assisted code review to anyone in
+an easy-to-setup, easy-to-use, easy-to-extend fashion. To achieve
+that, Lookout watches Github repos and triggers a set of analyzers
+when new code is pull-request-ed or pushed. Those analyzers are very
+easy to define (they are based on the gRPC toolsuite).
+
+In our case, when new code is pushed, we want to learn from a codebase
+to model its style as precisely as possible. When new code is
+pull-request-ed, we analyze it to detect any styling issue and report
```

Comment:
`to detect problems with style and suggest their fixes as code review comments. Particularly, we leverage [GitHub Suggested Changes](link)`